x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/configure_pipeline.tsx

```diff
@@ -173,14 +167,23 @@ export const ConfigurePipeline: React.FC = () => {
                 hasDividers
                 disabled={inputsDisabled}
                 itemLayoutAlign="top"
-                onChange={(value) =>
-                  setInferencePipelineConfiguration({
+                onChange={(value) => {
+                  const newConfiguration = {
                     ...configuration,
                     inferenceConfig: undefined,
                     modelID: value,
                     fieldMappings: undefined,
-                  })
-                }
+                    pipelineName: indexName + '-' + normalizeModelName(value),
+                    pipelineNameUserSupplied: false,
+                  };
+
+                  if (pipelineName && pipelineNameUserSupplied) {
+                    newConfiguration.pipelineName = pipelineName;
+                    newConfiguration.pipelineNameUserSupplied = true;
+                  }
```

Comment:
Nit: I _think_ we can simplify this, but please double check it:
```suggestion
                    pipelineName: pipelineNameUserSupplied ? pipelineName : indexName + '-' + normalizeModelName(value),
                    pipelineNameUserSupplied,
```

If so, we further simplify the return statement with the "return single object" notation:

```js
onChange={(value) => ({
  // list returned object's properties here, no need for return keyword
})}
```